lib/core/scheduler.rb

```diff
@@ -83,47 +80,7 @@ def sync_triggered?(connector_settings)
         return true
       end
 
-      # Don't sync if sync is explicitly disabled
-      scheduling_settings = connector_settings.scheduling_settings
-      unless scheduling_settings.present? && scheduling_settings[:enabled] == true
-        Utility::Logger.debug("#{connector_settings.formatted.capitalize} scheduling is disabled.")
-        return false
-      end
-
-      current_schedule = scheduling_settings[:interval]
-
-      # Don't sync if there is no actual scheduling interval
-      if current_schedule.nil? || current_schedule.empty?
-        Utility::Logger.warn("No sync schedule configured for #{connector_settings.formatted}.")
-        return false
-      end
-
-      current_schedule = begin
-        Utility::Cron.quartz_to_crontab(current_schedule)
-      rescue StandardError => e
-        Utility::ExceptionTracking.log_exception(e, "Unable to convert quartz (#{current_schedule}) to crontab.")
-        return false
-      end
-      cron_parser = Fugit::Cron.parse(current_schedule)
-
-      # Don't sync if the scheduling interval is non-parsable
-      unless cron_parser
-        Utility::Logger.error("Unable to parse sync schedule for #{connector_settings.formatted}: expression #{current_schedule} is not a valid Quartz Cron definition.")
-        return false
-      end
-
-      # We want to sync when sync never actually happened
-      last_synced = connector_settings[:last_synced]
-      if last_synced.nil? || last_synced.empty?
-        Utility::Logger.info("#{connector_settings.formatted.capitalize} has never synced yet, running initial sync.")
-        return true
-      end
-
-      next_trigger_time = cron_parser.next_time(Time.parse(last_synced))
-
-      # Sync if next trigger for the connector is in past
-      if next_trigger_time < Time.now
-        Utility::Logger.info("#{connector_settings.formatted.capitalize} sync is triggered by cron schedule #{current_schedule}.")
+      if schedule_triggered?(connector_settings.scheduling_settings, connector_settings.formatted, connector_settings.last_synced)
         return true
       end
 
```

Comment:
I think you can directly return schedule_triggered?, right?

```suggestion
     schedule_triggered?(connector_settings.scheduling_settings, connector_settings.formatted, connector_settings.last_synced)
```

Reply:
I can! Totally missed this, thanks